.github/workflows/wheels.yml

```diff
@@ -70,187 +70,98 @@ jobs:
+
+  build_wheels:
+    name: Build wheels on ${{ matrix.only }}
+    needs: generate-wheels-matrix
+    runs-on: ${{ matrix.os }}
 
-  non-Linux:
     strategy:
-      # Allows for matrix sub-jobs to fail without canceling the rest
       fail-fast: false
-
       matrix:
-        os: [macos-latest, windows-2019]
-        #os: [macos-10.15, windows-latest]
-        #os: [macos-10.15]
-        python-version:
-          - "2.7"
-          - "3.6"
-          - "3.7"
-          - "3.8"
-          - "3.9"
-          - "3.10"
-          - "3.11"
-          - "3.12"
-          - "pypy-3.8-v7.3.7"
-          - "pypy-3.9-v7.3.12"
-          - "pypy-3.10-v7.3.13"
-
-        #include:
-        #  - os: windows-2016
-        #    python-version: 2.7
-        exclude:
-          - os: windows-2019
-            python-version: 2.7  # needs older image
-
-    runs-on: ${{ matrix.os }}
-
-    permissions:
-      contents: write
-
-    env: { LIBXML2_VERSION: 2.11.5, LIBXSLT_VERSION: 1.1.38, MACOSX_DEPLOYMENT_TARGET: 11.0 }
+        include: ${{ fromJson(needs.generate-wheels-matrix.outputs.include) }}
 
     steps:
-    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-
-    - name: Set up Python
-      uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236 # v4.7.1
-      if: startsWith(matrix.python-version, '3.') || startsWith(matrix.python-version, 'pypy')
-      with:
-        python-version: ${{ matrix.python-version }}
-
-    - name: Install MacOS dependencies
-      if: startsWith(runner.os, 'mac')
-      run: |
-        brew install automake libtool
-        ln -s /usr/local/bin/glibtoolize /usr/local/bin/libtoolize
-
-    - name: Install dependencies
-      run: python -m pip install setuptools wheel -r requirements.txt
-
-    - name: Cache [libs]
-      uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
-      with:
-        path: |
-          libs/*.xz
-          libs/*.gz
-          libs/*.zip
-        key: libs-${{ runner.os }}
-
-    - name: Build wheels
-      run: make wheel
-      env: { STATIC_DEPS: true, RUN_TESTS: true }
-
-    - name: Release
-      uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844 # v0.1.15
-      if: startsWith(github.ref, 'refs/tags/')
-      with:
-        files: dist/lxml-*.whl
-
-    - name: Upload wheels
-      uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3.1.3
-      with:
-        name: wheels-${{ matrix.os }}
-        path: dist/lxml-*.whl
-        if-no-files-found: ignore
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        if: runner.os == 'Linux'
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: all
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.15.0
+        with:
+          only: ${{ matrix.only }}
+        env:
+          CIBW_SKIP: cp36-* cp37-* cp38-*
```

Comment:
I'd be happy to keep those included. Adding back Py2.7 support would also be nice, but we could also keep the old workflow around for that.

Reply:
Thank you for your comment. I will add all versions/arch that you already have. This is just the first trial. :) 

I opened PR since I hoped I would see build runs, but they do not run without approval, so currently, I am checking the runs on the branch on my fork.